sys/ztimer/periph_ptp.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C)    2018 Kaspar Schleiser <kaspar@schleiser.de>
+                    2021 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_ztimer_periph_ptp
+ * @{
+ *
+ * @file
+ * @brief       ztimer periph/ptp implementation
+ *
+ * @author      Jana Eisoldt <jana.eisoldt@ovgu.de>
+ 1
+ 1 @}
+ */
+#include "assert.h"
+
+#include "irq.h"
+#include "periph/ptp.h"
+#include "ztimer/periph_ptp.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static ztimer_clock_t *clock_timer;
+
+void ptp_timer_cb(void)
+{
+    ztimer_handler(clock_timer);
+}
+
+static void _ztimer_periph_ptp_set(ztimer_clock_t *clock, uint32_t val)
+{
+    (void) clock;
+
+    unsigned state = irq_disable();
+
+    clock_timer = clock;
+    ptp_timer_set_u64((uint64_t) val);
```

Comment:
```suggestion
    ptp_timer_set_u64(val);
```

Integer will be automatically promoted to higher types if the signature of the function require this. The cast is only needed when reducing the range of the integer type to silence warnings from the compiler.